VDA-StorageNode-Registry/contracts/StorageNodeRegistry.sol

```diff
@@ -0,0 +1,545 @@
+        uint unregisterDateTime,
+        bytes calldata requestSignature,
+        bytes calldata requestProof
+    ) external override {
+        uint nodeId = _didNodeId[didAddress];
+
+        // Check whether didAddress was registered before
+        if (nodeId == 0 || _nodeUnregisterTime[nodeId] != 0) {
+            revert InvalidDIDAddress();
+        }
+
+        if (unregisterDateTime < (block.timestamp + 28 days)) {
+            revert InvalidUnregisterTime();
+        }
+
+        bytes memory params = abi.encodePacked(didAddress, unregisterDateTime);
+        verifyRequest(didAddress, params, requestSignature, requestProof);
+
+        _nodeUnregisterTime[nodeId] = unregisterDateTime;
+
+        emit RemoveNodeStart(didAddress, unregisterDateTime);
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function removeNodeComplete(
+        address didAddress,
+        bytes calldata requestSignature,
+        bytes calldata requestProof
+    ) external override {
+        uint nodeId = _didNodeId[didAddress];
+        {
+            if (nodeId == 0) {
+                revert InvalidDIDAddress();
+            }
+
+            if (_nodeUnregisterTime[nodeId] == 0 || _nodeUnregisterTime[nodeId] > block.timestamp) {
+                revert InvalidUnregisterTime();
+            }
+
+            bytes memory params = abi.encodePacked(didAddress);
+            verifyRequest(didAddress, params, requestSignature, requestProof);
+        }
+
+        StorageNode storage nodeInfo = _nodeMap[nodeId];
+
+        --_datacenterInfo[nodeInfo.datacenterId].nodeCount;
+
+        _countryNodeIds[nodeInfo.countryCode].remove(nodeId);
+        _regionNodeIds[nodeInfo.regionCode].remove(nodeId);
+        delete _endpointNodeId[nodeInfo.endpointUri];
+        delete _didNodeId[didAddress];
+        delete _nodeMap[nodeId];
+
+        delete _nodeUnregisterTime[nodeId];
+
+        emit RemoveNodeComplete(didAddress);
+    }
+
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function getNodeByAddress(address didAddress) external view override returns(StorageNode memory) {
+        uint nodeId = _didNodeId[didAddress];
+
+        if (nodeId == 0) {
+            revert InvalidDIDAddress();
+        }
+
+        if (_nodeUnregisterTime[nodeId] != 0) {
+            revert InvalidDIDAddress();
+        }
+
+        return _nodeMap[nodeId];
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function getNodeByEndpoint(string calldata endpointUri) external view override returns(StorageNode memory) {
+        uint nodeId = _endpointNodeId[endpointUri];
+
+        if (nodeId == 0) {
+            revert InvalidEndpointUri();
+        }
+
+        if (_nodeUnregisterTime[nodeId] != 0) {
+            revert InvalidEndpointUri();
+        }
+
+        return _nodeMap[nodeId];
+    }
+
+    function filtrActiveStorageNodes(EnumerableSetUpgradeable.UintSet storage ids) internal view returns(StorageNode[] memory) {
```

Comment:
Typo, should be `filterActiveStorageNodes`

Reply:
Updated.